File: JavaScript solutions/342. Power of Four.js
/**
 * @param {number} n
 * @return {boolean}
 */
var isPowerOfFour = function(n) {
    var x=0;
    while(1){
        if(Math.pow(4,x)==n) return true;
        if(Math.pow(4,x)>n) return false;
        x++;
    }
};

File: C++ solutions/342. Power of Four.cpp
#include<bits/stdc++.h>
using namespace std;

// solution
class Solution {
public:
    bool isPowerOfFour(int n) {
        int x=0;
        while(1){
            if(pow(4,x)==n) return true;
            if(pow(4,x)>n) return false;
            x++;
        }
    }
};
